### ISSUE TITLE
Update Mockk Dependency to 1.12.1

### ISSUE THREAD

**Issue by sandrozbinden** [Ninja-Squad/springmockk#68]
The current mockk dependency (`1.10.2`) has issues with kotlin inline classes. Would it be possible to update to the current `1.12.1` dependency?

**Comment by cexbrayat** [comment]
Hi @sandrozbinden 

Thank you for opening an issue. This is a fair point! We'll look into it in the next few days, and probably cut a release with the updated dependencies 👍

**Comment by jnizet** [comment]
Hi @sandrozbinden 

I will do an upgrade because... why not.
But you should know that you don't need to wait for SpringMockK to upgrade in order to use a more recent, compatible version of MockK. All you need to do is declare a dependency to the MockK version you want inside your own project.

WIth gradle, it's as simple as

```
dependencies {
    // ...
    testImplementation("com.ninja-squad:springmockk:3.0.1")
    testImplementation("io.mockk:mockk:1.12.1")
}
```

**Comment by jnizet** [comment]
I've released version 3.1.0
